### ISSUE TITLE
Selector stopped working inside MIGXdb configs

### ISSUE THREAD

**Issue by hugopeek** [Jako/SuperBoxSelect#30]
After updating to 3.0.0, I can't use the selectors anymore inside MIGX modals. Contents are not being loaded and the follow errors emerge:

```
[2022-01-15 11:34:16] (ERROR @ /home/hugo/Localhost/releaf-aid/core/components/superboxselect/elements/tv/input/superboxselect.class.php : 75) PHP warning: Invalid argument supplied for foreach()
[2022-01-15 11:34:16] (ERROR @ /home/hugo/Localhost/releaf-aid/core/components/superboxselect/elements/tv/input/superboxselect.class.php : 75) PHP warning: Invalid argument supplied for foreach()
[2022-01-15 11:34:16] (ERROR in SuperBoxSelect @ /home/hugo/Localhost/releaf-aid/core/components/superboxselect/processors/types/resources/getlist.class.php : 44) Invalid template variable ID!
[2022-01-15 11:34:16] (ERROR in SuperBoxSelect @ /home/hugo/Localhost/releaf-aid/core/components/superboxselect/processors/types/users/getlist.class.php : 44) Invalid template variable ID!
```

This is the MIGX config, which works in 2.X:
```json
{
    "MIGX_id": "",
    "field": "contact_id",
    "caption": "Contact",
    "description": "Optional. If the source is already registered on this website as a user, you can select him/her here. No need to fill in contact details if selected.",
    "description_is_code": "0",
    "inputTV": "",
    "inputTVtype": "superboxselect",
    "validation": "",
    "configs": {
        "selectType": "users",
        "maxElements": "1",
        "allowBlank": "1",
        "pageSize": "20"
    },
    "restrictive_condition": "",
    "display": "",
    "sourceFrom": "config",
    "sources": "",
    "inputOptionValues": "",
    "default": "",
    "useDefaultIfEmpty": "0",
    "pos": ""
}
```

**Comment by Jako** [comment]
You have to replace the following lines in `core/components/superboxselect/processors/types/users/getlist.class.php` and in `core/components/superboxselect/processors/types/resources/getlist.class.php` in the method `prepareQueryBeforeCount`:

```
        if ($tv) {
            $tvProperties = $tv->get('input_properties');
        } elseif ($this->getProperty('useRequest', 0)) {
            $tvProperties = $this->getProperties();
        } else {
            $tvProperties = [];
            $c->where(['id' => 0]);
            $this->modx->log(xPDO::LOG_LEVEL_ERROR, 'Invalid template variable ID!', '', 'SuperBoxSelect');
        }
```

and add `"useRequest": "1"` to the MIGX config `configs` object.

I did not want to run the connector with query properties and in MIGX there is no TV set in your case.

Is there a documentation for the SuperBoxSelect usage in MIGX, where this information can be added?

**Comment by hugopeek** [comment]
Thanks for the quick reply @Jako. I added the elseif statement and the useRequest property, but it's not working unfortunately.

It seems the config object is not being picked up by the properties array at all:
```
[2022-01-15 15:26:11] (ERROR in SuperBoxSelect @ /home/hugo/Localhost/pff/core/components/superboxselect/processors/types/users/getlist.class.php : 46) User props: Array
(
    [start] => 0
    [limit] => 20
    [sort] => username
    [dir] => ASC
    [combo] => 
    [query] => 
    [id] => 
    [action] => types/users/getlist
    [tvid] => inp_1924_2_4
    [resourceId] => 0
    [contextKey] => web
)
```

**Comment by Jako** [comment]
It was somehow more work to fix that: Please try the changes in https://github.com/Jako/SuperBoxSelect/commit/aa056a1872fcc7abc8695bec60fda68289afb310. There is an installable package too with the changes.

**Comment by Jako** [comment]
@hugopeek Did you get a chance to test the new version? https://github.com/Jako/SuperBoxSelect/raw/aa056a1872fcc7abc8695bec60fda68289afb310/_packages/superboxselect-3.0.1-rc1.transport.zip

**Comment by hugopeek** [comment]
@Jako Sorry for the delay.. I just installed it and it seems to work for both resources and users now. Thanks a bunch!!

One warning remains:

```
[2022-01-20 12:54:39] (ERROR @ /home/hugo/Localhost/pff/core/components/superboxselect/elements/tv/input/superboxselect.class.php : 77) PHP warning: Invalid argument supplied for foreach()
```

And do you still need to add `"useRequest": "1"` to the config? It seems to be fine without now..

**Comment by Jako** [comment]
It works without any custom property filtering the list like parents, depth, where etc. out of the box. The warning is fixed in 3.0.1 too.
